exchanges/binance/binance_test.go

```diff
@@ -1477,109 +1477,108 @@ func TestNewOrderTest(t *testing.T) {
 func TestGetHistoricTrades(t *testing.T) {
 	t.Parallel()
 	p := currency.NewBTCUSDT()
-	start := time.Unix(1577977445, 0)  // 2020-01-02 15:04:05
-	end := start.Add(15 * time.Minute) // 2020-01-02 15:19:05
-	result, err := e.GetHistoricTrades(t.Context(), p, asset.Spot, start, end)
-	assert.NoError(t, err, "GetHistoricTrades should not error")
-	expected := 2134
+	start := time.Now().Add(-time.Hour * 24 * 90).Truncate(time.Minute) // 3 months ago
 	if mockTests {
-		expected = 1002
+		start = time.Unix(1577977445, 0) // 2020-01-02 15:04:05
 	}
-	assert.Equal(t, expected, len(result), "GetHistoricTrades should return correct number of entries")
+	end := start.Add(15 * time.Minute)
+	result, err := e.GetHistoricTrades(t.Context(), p, asset.Spot, start, end)
+	require.NoError(t, err, "GetHistoricTrades must not error")
+	assert.Greater(t, len(result), 1001, "GetHistoricTrades should enough trades")
 	for _, r := range result {
-		if !assert.WithinRange(t, r.Timestamp, start, end, "All trades should be within time range") {
-			break
-		}
+		require.WithinRange(t, r.Timestamp, start, end, "All trades must be within time range")
 	}
 }
 
+// TestGetAggregatedTradesBatched exercises TestGetAggregatedTradesBatched to ensure our date and limit scanning works correctly
+// This test is susceptible to failure if volumes change a lot, during wash trading or zero-fee periods
+// In live tests, 45 minutes is expected to return more than 1000 records
 func TestGetAggregatedTradesBatched(t *testing.T) {
```

Comment:
this fails on a live test

Reply:
Thanks!
Could you be more specific?
Did it fail locally for you, and can you replicate now?

```
❯ go test -count=1 --tags mock_test_off -v  ./exchanges/binance/ -run TestGetAggregatedTradesBatched/
--- PASS: TestGetAggregatedTradesBatched (0.00s)
    --- PASS: TestGetAggregatedTradesBatched/custom_limit_with_start_time_set,_no_end_time (0.46s)
    --- PASS: TestGetAggregatedTradesBatched/batch_with_timerange (4.77s)
PASS
ok      github.com/thrasher-corp/gocryptotrader/exchanges/binance       8.984s
```